include/net/net_mgmt.h

```diff
@@ -189,8 +193,24 @@ void net_mgmt_del_event_callback(struct net_mgmt_event_callback *cb);
  * @param mgmt_event The actual network event code to notify
  * @param iface a valid pointer on a struct net_if if only the event is
  *        based on an iface. NULL otherwise.
+ * @param info a valid pointer on the information you want to pass along
+ *        with the event. NULL otherwise. Note the data pointed there is
+ *        normalized by the related event.
+ * @param length size of the data pointed by info pointer.
+ *
+ * Note: info and length are disabled if CONFIG_NET_MGMT_EVENT_INFO
+ *       is not defined.
  */
-void net_mgmt_event_notify(u32_t mgmt_event, struct net_if *iface);
+#ifdef CONFIG_NET_MGMT_EVENT_INFO
+void net_mgmt_event_notify(u32_t mgmt_event, struct net_if *iface,
```

Comment:
I think it is better to call this net_mgmt_event_notify_info() or similar, so that we can avoid having ifdefs in the code that  uses this function. We could even make the net_mgmt_event_notify() to call the new function with NULL info.

Reply:
hum, naming will be an issue here. It's an event we are notifying, not something else.

What about:
- net_mgmt_event_notify()  with info parameter
- net_mgmt_event_notify_only()  or net_mgmt_event_notify_no_info() ? for the non-info based notification.

Reply:
actually:
- net_mgmt_event_notify()  no info parameter
- net_mgmt_event_notify_with_info() 

sound better, less changes.